paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -304,6 +328,32 @@ internal abstract class BaseSheetViewModel<TransitionTargetType>(
         logger.warning(message)
     }
 
+    fun updatePrimaryButtonText(text: String, initial: Boolean = false) {
+        if (initial) {
+            initialPrimaryButtonText = text
+        }
+        _primaryButtonText.value = text
+    }
+
+    fun updatePrimaryButtonEnabled(enabled: Boolean) {
+        primaryButtonEnabled.value = enabled
+    }
+
+    fun updatePrimaryButtonOnPress(initial: Boolean = false, onPress: () -> Unit) {
```

Comment:
Do we really need the `initial` parameter here and the special logic? Just wondering if we can make it simpler, if not please just add comments

Reply:
Cleaned this up, let me know what you think